src/styles/_header.scss

```
@@ -37,5 +37,11 @@
     text-transform: uppercase;
     font-size: 12px;
     font-weight: 500;
+    transition-duration: 300ms;
+    transition-property: transform;
+
+    &:hover {
+      color: #00acdc;
+    }
```

Здесь ты анимируешь не transform, а color
```suggestion
    transition-duration: 300ms;
    transition-property: color;

    &:hover {
      color: #00acdc;
    }
```
Либо можно записать одним свойством: transition: color 300ms;